node/adapter/ipam.py

```diff
@@ -0,0 +1,115 @@
+from netaddr import IPAddress
+from datastore import DatastoreClient
+
+IP_ASSIGNMENT_PATH = "/calico/ipam/%(version)s/assignment/%(pool)s"
+IP_ASSIGNMENT_KEY = IP_ASSIGNMENT_PATH + "/%(address)s"
+
+
+class SequentialAssignment(object):
+    """
+    Assign IP addresses sequentially
+    """
+
+    def __init__(self):
+        # Init an etcd client.
+        self.etcd = IPAMClient()
+
+    def allocate(self, pool):
+        """
+        Attempt to allocate an IP address from the provided pool.
+
+        :param IPNetwork pool: The pool to allocate from
+        :return: An IP address which has been allocated or None
```

Comment:
Again, throw an exception if we fail.  It's really easy to write didn't-check-the-return-value bugs that are hard to diagnose.